### PR TITLE
Introduce runtime build type detection to replace compile-time flags

### PR DESCRIPTION
**Pull request by mallexxx**
Task/Issue URL: https://app.asana.com/0/0/1213329987846867
Tech Design URL: N/A
CC: N/A

### Description

- Added `AppVersion.isAppStoreBuild` static property to `Common` package for universal App Store build detection across all targets based on bundle id
- Introduced `isAlphaBuild` and `isReviewBuild` to `ApplicationBuildType` protocol with bundle identifier-based runtime detection (`.alpha` and `.review` suffixes)
- Replaced compile-time `#if APPSTORE` checks with runtime `AppVersion.isAppStoreBuild` across multiple files including menus, preferences, and feedback
- Fixed `.keyboardShortcut` availability issue in `DataImportView` using `.ifLet` pattern for macOS 12 compatibility
- Updated mock implementations (`ApplicationBuildTypeMock`) to include new protocol requirements

### Testing Steps

1. **Perform smoke testing on all build configurations** (Debug, Release, App Store, Review, Alpha)
2. **Verify runtime behavior:**
   - Run DMG build and verify "About" panel shows alpha label when appropriate (alpha build only)
   - Run App Store build and verify no alpha label appears in non-alpha configurations
   - Verify `AppVersion.isAppStoreBuild` returns correct values in debugger across different build configurations

### Impact and Risks

**Impact Level: Medium**

#### What could go wrong?

- **Bundle identifier changes**: If bundle ID naming conventions change in the future, runtime checks could break
- **Test infrastructure**: Mocks need to be kept in sync with protocol changes

**Mitigation:**
- Bundle ID patterns are well-established and consistent across all targets
- Runtime checks are simple string prefix/suffix checks with safe fallbacks (`?? false`)
- Existing test coverage validates the behavior
- This is foundation work - subsequent PRs will complete the migration by removing actual flag definitions from xcconfig files

### Quality Considerations

- **Edge cases**: Runtime checks handle missing bundle identifiers gracefully with `?? false` fallbacks
- **Performance**: Bundle identifier checks are cached as static properties (evaluated once at launch)
- **Monitoring**: No new analytics, existing pixel infrastructure remains unchanged
- **Documentation**: Code comments explain runtime detection approach in `AppVersion` and `ApplicationBuildType`
- **Privacy/Security**: No privacy implications; changes are purely build configuration detection

### Notes to Reviewer

This is foundational work for dropping compile-time flags. This PR introduces the **runtime detection infrastructure** (`AppVersion.isAppStoreBuild`, enhanced `ApplicationBuildType`) and migrates many call sites to use it.

The actual removal of compile-time flag definitions from xcconfig files will come in a follow-up PR, ensuring we can validate the runtime approach works correctly first.

Key architectural decision: Placing `isAppStoreBuild` in `AppVersion` (Common package) rather than only in `ApplicationBuildType` (main app) ensures it's accessible from system extensions and agent targets that don't have access to the main app's `ApplicationBuildType` protocol.

Made with [Cursor](https://cursor.com)

---

> [!NOTE]
> **Medium Risk**
> Medium risk because many code paths previously gated by `#if APPSTORE`/`#if SPARKLE`/`#if REVIEW` now depend on runtime bundle-id/sandbox detection, which could change behavior across distributions and affect update flow, keychain account selection, and debug-only tooling.
> 
> **Overview**
> This PR replaces a broad set of compile-time build flags (`APPSTORE`, `SPARKLE`, `REVIEW`, `ALPHA`) with **runtime build-type detection** based on bundle identifier and sandbox status.
> 
> It adds `AppVersion.isAppStoreBuild` (bundle-id prefix check) and extends `ApplicationBuildType` with runtime `isReviewBuild`/`isAlphaBuild`, then updates many call sites (menus, preferences, onboarding/config env selection, pixels/feedback, automation/debug menus) to use `StandardApplicationBuildType`/`NSApp.isSandboxed` instead of conditional compilation.
> 
> It also adjusts a few behavioral edge cases tied to distribution (e.g., keychain account selection in `EncryptionKeyStore`, PiP and file-URL restrictions, download-location security-scoped bookmarks) and updates local-package clients (`HistoryView`/`NewTabPage`) to take an explicit `environment` parameter (with tests updated).
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit f661aed1bf3b4be447b003c2cfb8d1283a62335c. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: macOS/DuckDuckGo/Menus/MainMenu.swift [hunk 1]
```diff
@@ -1010,10 +1003,9 @@ final class MainMenu: NSMenu {
         debugMenu.insertItem(separatorItem, at: 1)
 
         debugMenu.addItem(internalUserItem)
-#if !ALPHA
         debugMenu.addItem(.separator())
         debugMenu.addItem(NSMenuItem(title: "Download DuckDuckGo Alpha Build", action: #selector(downloadAlphaBuild), target: self))
```
Comment (mallexxx) [inline macOS/DuckDuckGo/Menus/MainMenu.swift:1007]: we should probably be fine with the link in the alpha build

File: macOS/DuckDuckGo/Feedback/Model/FeedbackSender.swift [hunk 2]
```diff
@@ -32,16 +33,12 @@ final class FeedbackSender: FeedbackSenderImplementing {
     static let feedbackURL = URL.feedbackForm
 
     func sendFeedback(_ feedback: Feedback, completionHandler: (() -> Void)? = nil) {
-#if DEBUG || REVIEW
-        Logger.general.debug("FeedbackSender: Skipping feedback submission in DEBUG / REVIEW build")
+#if DEBUG
+        Logger.general.debug("FeedbackSender: Skipping feedback submission in DEBUG build")
```
[the remark this answers is not in the document]
Comment (mallexxx) [reply macOS/DuckDuckGo/Feedback/Model/FeedbackSender.swift:37]: the feedback form has been tested so I guess it's fine that reports can be send from the REVIEW build

File: macOS/DuckDuckGo/Tab/TabExtensions/PopupHandlingTabExtension.swift [hunk 3]
```diff
@@ -333,13 +333,11 @@ final class PopupHandlingTabExtension {
     func isNavigationActionUserInitiated(_ navigationAction: WKNavigationAction) -> UserInitiatedReason? {
         var threshold = popupBlockingConfig.userInitiatedPopupThreshold
 
-#if DEBUG || REVIEW
```
[the remark this answers is not in the document]
Comment (mallexxx) [reply macOS/DuckDuckGo/Tab/TabExtensions/PopupHandlingTabExtension.swift:336]: not a big deal I guess…

File: macOS/DuckDuckGo/Automation/LaunchOptionsHandler.swift [hunk 4]
```diff
@@ -47,24 +47,22 @@ public final class LaunchOptionsHandler {
...
 
     /// Returns true if the app is running in any automation mode (WebDriver or UI Tests)
     public var isAutomationSession: Bool {
-#if DEBUG || REVIEW
-        isWebDriverAutomationSession || isUITesting
-#else
-        isUITesting
-#endif
+        let buildType = StandardApplicationBuildType()
+        guard buildType.isDebugBuild || buildType.isReviewBuild else { return false }
```
Comment (ayoy) [inline macOS/DuckDuckGo/Automation/LaunchOptionsHandler.swift]: This should be returning `isUITesting`, not `false`.
